packages/flutter/lib/src/material/data_table.dart

```diff
@@ -569,6 +574,15 @@ class DataTable extends StatelessWidget {
   /// margin between the checkbox and the content in the first data column.
   final double? horizontalMargin;
 
+  /// {@template flutter.material.dataTable.topBottomRowPadding}
+  /// The minimum padding between the content of each data cell and the top and bottom of the data row.
+  /// 
+  /// If null, [DataTableThemeData.topBottomRowPadding] is used. This value
+  /// defaults to 0.0.
+  /// 
+  /// Also see [DataTable.dataRowHeight].
+  final double? topBottomRowPadding;
```

Comment:
Alternative:
A disadvantage of the solution is that there is a dependency between dataRowHeight and topBottomRowPadding (when increasing dataRowHeight you will need to also increase topBottomRowPadding if there are rows with height larger than dataRowHeight). An alternative that would be more self-explanatory for the user would be to replace the parameters dataRowHeight and topBottomRowPadding with a parameter that can be either a dataRowHeight or a topBottomRowPadding.
If topBottomRowPadding is specified then dataRowHeight would be set to 0.0 and if dataRowHeight is set then the existing behaviour of an overflow would occur.
The dataRowHeight parameter would need to be depreciated because it would be replaced by the new choice parameter.

Reply:
I think alternative 1 is more prefered, although I think it doesn't need to be a new data class. you can create a new parameter for dynamic row height, and assert one one is passed into the data table constructor

Reply:
With the latest commits, I chose to create the type DataTableRowHeight for the dataRowHeightSettings parameter because then it is easier for the user to discover all possible DataRow height settings.
Please review the current changes again.